includes/Checker/Checks/Plugin_Repo/File_Type_Check.php

```diff
@@ -302,6 +302,49 @@ function ( $file ) use ( $plugin_path ) {
 				);
 			}
 		}
+
+		// Duplicated names in different folders.
+		$folders = array();
+		foreach ( $files as $file ) {
+			$folder = str_replace( basename( $file ), '', $file );
+			if ( empty( $folder ) ) {
+				continue;
+			}
+			$folders[] = $folder;
+		}
+		$folders                = array_unique( $folders );
+		$folders_lowercase      = array_map( 'strtolower', $folders );
+		$case_sensitive_folders = array_unique( array_diff_assoc( $folders_lowercase, array_unique( $folders_lowercase ) ) );
+
+		if ( ! empty( $case_sensitive_folders ) ) {
+			$this->add_result_error_for_file(
+				$result,
+				__( 'Case-Sensitive duplicated folders were found. They\'ll be problematic in different systems.', 'plugin-check' ),
```

Comment:
```suggestion
				__( 'Multiple folders with the same name but different case were found. This can be problematic on certain file systems.', 'plugin-check' ),
```